internal/impl/devbox.go

```diff
@@ -125,6 +125,7 @@ func Open(path string, writer io.Writer) (*Devbox, error) {
 	if err != nil {
 		return nil, err
 	}
+	box.pluginManager.ApplyOptions(plugin.WithLockfile(lock))
```

Comment:
I'd suggest doing:

1. remove `pluginmanager: plugin.NewManager(),` above
2. make this `box.pluginManager = plugin.NewManager(plugin.WithLockfile(lock))` 
3. make `applyOptions` an internal function, 

(3) would make NewManager a more genuine constructor. That would make it easier to reason about that all the pluginManager's fields are ready and methods can be applied.